Properly correct percent literal delimiters with escape characters in them

rrosenblum opened the pull request:
I had an issue where something like `%{\n}` would be corrected to
```ruby
%(
)
```
While trying to fix this, it became easier to replace the delimiter than to figure proper escaping. I am pretty happy with the simplicity of this fix.

bbatsov commented on spec/rubocop/cop/style/percent_literal_delimiters_spec.rb:
```diff
@@ -317,5 +317,32 @@
     end
+
+    shared_examples :escape_characters do |percent_literal|
+      it "corrects #{percent_literal} with \\n as a symbol" do
```
As a symbol?

rrosenblum replied:
Whoops, old naming.

bbatsov commented:
Hmm, the changes look reasonable, although I wonder if @jonas054 has something more in mind with the old code.

jonas054 commented:
No, I don't think I did. All the old spec examples still pass after this change, and it doesn't look like it's due to poor coverage. It's just that my solution was needlessly complicated. Great work, @rrosenblum!

bbatsov commented on CHANGELOG.md:
```diff
@@ -39,6 +39,7 @@
 * [#2059](https://github.com/bbatsov/rubocop/issues/2059): Don't check for trivial accessors in modules. ([@bbatsov][])
 * Add proper punctuation to the end of offense messages, where it is missing. ([@lumeet][])
 * [#2071](https://github.com/bbatsov/rubocop/pull/2071): Keep line breaks in place on WordArray autocorrect.([@unmanbearpig][])
+* Properly correct `Style/PercentLiteralDelimiters` with escape characters in them. ([@rrosenblum][])
```
Mention the PR here for completeness.

rrosenblum commented:
Thanks, it took me a while to realize all that we needed to do was replace the delimiter. I originally was trying to add escaping to the escape characters which was proving difficult.

I have updated the code and rebased.

bbatsov commented:
Hmm, the build failed.

rrosenblum commented:
It looks like the build failed during bundle install. Not sure why that would happen here. I triggered another build so we will see what happens.

rrosenblum commented:
Build succeeded this time.

rrosenblum commented:
Speaking of the build, we still have failures in Jruby and RBX.

bbatsov commented:
Yeah, we should address them one of these days...
